common/script_values/wc_men_at_arms_values.txt

```diff
@@ -83,7 +83,7 @@ culture_ai_weight_caster = {
 	}
 }
 culture_ai_weight_massive_infantry = {
-	value = -18
+	value = 300
```

Comment:
Is there a formula behind these values?

Reply:
I compared their stats/cost to vanilla units so they are not too op or weak, and I made the costs appropriate.  I didn't use formula because any formula I would use would cause massive balance gaps between units.   For example, the siege weapons are super low in terms of stats, but Paradox they inflated their value to the ai, so i did the same with these units.  Their stats would be only balanced in theory, until there are LOTS of simulations happening.

Reply:
> I compared their stats/cost to vanilla units so they are not too op or weak, and I made the costs appropriate. I didn't use formula because any formula I would use would cause massive balance gaps between units. For example, the siege weapons are super low in terms of stats, but Paradox they inflated their value to the ai, so i did the same with these units. Their stats would be only balanced in theory, until there are LOTS of simulations happening.

Paradox uses the formula for the AI unit picking. I linked it somewhere. You can iterate through cost and AI values, but that's like finding a way out of a labyrinth blindfolded. You better crack the formula and understand how it works.